### PR TITLE
Document cli options that can be set in configuration

### PR DESCRIPTION
**Pull request by irkode**
implements [Forum Topic 51064](https://discourse.gohugo.io/t/disablefastrender-in-the-config-file/51064/2) 

- add missing settings
- fix sort order of headings in "All configuration settings"

### CODE REVIEW

**Comment by irkode** [comment]
Remark:  regarding sort order of Section[All configuration settings](https://gohugo.io/getting-started/configuration/#all-configuration-settings)

It looks like all these list should be sorted which imho is a good setup. 
Currently four entries are not at sort order location.

- canonifyURLs
- enableRobotsTXT
- disableLanguages
- removePathAccents

if you want I could add another commit to align the sort (or create another PR or leave as is)  just as you prefer.

**Comment by jmooring** [comment]
Thanks. Please rebase and make the sort corrections you mentioned.

**Comment by irkode** [comment]
rebased, sorted, updated PR and set to Review

File: content/en/getting-started/configuration.md [hunk 1]
```diff
@@ -312,6 +316,10 @@ To remain consistent and prevent unexpected behavior, do not mix these strategie
 
 (`bool`) Enable generation of `robots.txt` file. Default is `false`.
 
+###### environment
+
+(`string`) Build environment. Default is _production_ when called as  `hugo` and _development_ when called as `hugo server`. See [Configuration directory](https://gohugo.io/getting-started/configuration/#configuration-directory)
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:321]:
```suggestion
(`string`) Build environment. Default is `production` when running`hugo` and `development` when running `hugo server`. See [Configuration directory](https://gohugo.io/getting-started/configuration/#configuration-directory).
```

File: content/en/getting-started/configuration.md [hunk 2]
```diff
@@ -320,6 +328,14 @@ See [Front matter Configuration](#configure-front-matter).
 
 (`bool`) If true, auto-detect Chinese/Japanese/Korean Languages in the content. This will make `.Summary` and `.WordCount` behave correctly for CJK languages. Default is `false`.
 
+###### ignoreCache
+
+(`bool`) Ignores the cache directory. Default is `false`.
+
+###### ignoreVendorPaths
+
+(`string`) ignores any _vendor for module paths matching the given Glob pattern. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:337]:
```suggestion
(`string`) Ignore vendored modules that match the given [Glob] pattern within the `_vendor` directory .

[Glob]: https://github.com/gobwas/glob?tab=readme-ov-file#example]
```

File: content/en/getting-started/configuration.md [hunk 3]
```diff
@@ -288,6 +288,10 @@ To remain consistent and prevent unexpected behavior, do not mix these strategie
 
 [kinds]: /getting-started/glossary/#page-kind
 
+###### disableLanguages
+
+See [Disable a Language](/content-management/multilingual/#disable-a-language)
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:293]:
```suggestion
See [disable a language](/content-management/multilingual/#disable-a-language).
```

File: content/en/getting-started/configuration.md [hunk 4]
```diff
@@ -320,6 +328,14 @@ See [Front matter Configuration](#configure-front-matter).
 
 (`bool`) If true, auto-detect Chinese/Japanese/Korean Languages in the content. This will make `.Summary` and `.WordCount` behave correctly for CJK languages. Default is `false`.
 
+###### ignoreCache
+
+(`bool`) Ignores the cache directory. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:333]:
```suggestion
(`bool`) Ignore the cache directory. Default is `false`.
```

File: content/en/getting-started/configuration.md [hunk 5]
```diff
@@ -338,9 +354,9 @@ When present in the root of the configuration, this value is ignored if one or m
 
 See [Configure Languages](/content-management/multilingual/#configure-languages).
 
-###### disableLanguages
+###### layoutDir
 
-See [Disable a Language](/content-management/multilingual/#disable-a-language)
+(`string`) The directory where Hugo reads the layouts from. Default is `layouts`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:359]:
```suggestion
(`string`) The directory that contains templates. Default is `layouts`.
```

File: content/en/getting-started/configuration.md [hunk 6]
```diff
@@ -386,6 +406,10 @@ See [configure page](#configure-page).
 
 See [configure pagination](/templates/pagination/#configuration).
 
+###### panicOnWarning
+
+(`bool`) Panic on first WARNING log. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:411]:
```suggestion
(`bool`) Whether to panic on first the WARNING log. Default is `false`.
```

File: content/en/getting-started/configuration.md [hunk 7]
```diff
@@ -394,6 +418,18 @@ See [Content Management](/content-management/urls/#permalinks).
 
 (`bool`) Whether to pluralize automatic list titles. Applicable to section pages. Default is `true`.
 
+###### printI18nWarnings
+
+(`bool`) Print missing translations. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:423]:
```suggestion
(`bool`) Whether to print missing translations to the console. Default is `false`.
```

File: content/en/getting-started/configuration.md [hunk 8]
```diff
@@ -394,6 +418,18 @@ See [Content Management](/content-management/urls/#permalinks).
...
+
+###### printPathWarnings
+
+(`bool`) Print warnings on duplicate target paths etc. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:427]:
```suggestion
(`bool`) Whether to print warnings to the console when Hugo publishes two or more files to the same path. Default is `false`.
```

File: content/en/getting-started/configuration.md [hunk 9]
```diff
@@ -394,6 +418,18 @@ See [Content Management](/content-management/urls/#permalinks).
...
+
+###### printUnusedTemplates
+
+(`bool`) Print warnings on unused templates. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:431]:
```suggestion
(`bool`) Whether to print a warning to the console for each unused template. Default is `false`.
```

File: content/en/getting-started/configuration.md [hunk 10]
```diff
@@ -455,6 +491,15 @@ Default [sitemap configuration](/templates/sitemap/#configuration).
 
 See [Configure Taxonomies](/content-management/taxonomies#configure-taxonomies).
 
+###### templateMetrics
+
+(`bool`) Display metrics about template executions. Default is `false`.
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:496]:
```suggestion
(`bool`) Whether to print template execution metrics to the console. Default is `false`.
```

File: content/en/getting-started/configuration.md [hunk 11]
```diff
@@ -455,6 +491,15 @@ Default [sitemap configuration](/templates/sitemap/#configuration).
...
+See [Template metrics](/troubleshooting/performance/#template-metrics)
+
+###### templateMetricsHints
+
+(`bool`) Calculate some improvement hints when _templateMetrics_ option `true`. Default is `false`. See [Template metrics](/troubleshooting/performance/#template-metrics)
```
Comment (jmooring) [inline content/en/getting-started/configuration.md:501]:
```suggestion
(`bool`) Whether to print template execution improvement hints to the console. Applicable when `templateMetrics` is `true`. Default is `false`. See [template metrics](/troubleshooting/performance/#template-metrics).
```

**Review by jmooring — CHANGES_REQUESTED** [review]

File: content/en/getting-started/configuration.md (hunk 6 above)
Comment (irkode) [reply content/en/getting-started/configuration.md:411]: Suggestions:
1. Whether to panic on first WARNING. 
2. Whether to panic on the first WARNING.

File: content/en/getting-started/configuration.md (hunk 7 above)
Comment (irkode) [reply content/en/getting-started/configuration.md:423]: my suggestion - see global comment:
  (`bool`) Whether to log WARNINGs for each missing translation. Default is `false`.

File: content/en/getting-started/configuration.md (hunk 8 above)
Comment (irkode) [reply content/en/getting-started/configuration.md:427]: my suggestion - see global comment:
(`bool`)  Whether to log WARNINGs when Hugo publishes two or more files to the same path. Default is `false`.

File: content/en/getting-started/configuration.md (hunk 9 above)
Comment (irkode) [reply content/en/getting-started/configuration.md:431]: my suggestion - see global comment:
(`bool`)  Whether to log WARNINGs for each unused template. Default is `false`.

**Comment by irkode** [comment]
Thx for the review - I just took the stuff literally from the comandline help



What about using the phrase `Whether to log WARNINGs when ...` for the --print... flags
- Would match the description in `fmt.warnf`.
-  imho makes using _to the console_ superfluous

maybe without capitalizing warnings.

- printI18nWarnings (this actually is a warning not just additional output)
  (`bool`) Whether to log WARNINGs for each missing translation. Default is `false`.

- printPathWarnings
  (`bool`)  Whether to log WARNINGs when Hugo publishes two or more files to the same path. Default is `false`.

- printUnusedTemplates
  (`bool`)  Whether to log WARNINGs for each unused template. Default is `false`.

**Comment by jmooring** [comment]
> What about using the phrase

Sure.

File: content/en/getting-started/configuration.md (hunk 6 above)
Comment (irkode) [reply content/en/getting-started/configuration.md:412]: What about that last one? I would use:
>Whether to panic on first WARNING.

**Comment by irkode** [comment]
hope that finally fits

**Pull request merged**